biz.aQute.bndlib/src/aQute/bnd/osgi/Clazz.java

```diff
@@ -1928,6 +1931,30 @@ public boolean is(QUERY query, Instruction instr, Analyzer analyzer) throws Exce
 
 				return false;
 
+			case META_ANNOTATED :
+				if (annotations == null)
+					return false;
+				Stack<TypeRef> stack = new Stack<>();
```

Comment:
Don't use Stack. It is an old type with excessive synchronization. Use a Dequeue implementation like ArrayDeque instead.

Reply:
Actually, what is the need for a stack anyway? Is the order you visit the annotation on a type of any meaning?

Reply:
recursion without method calls?